### PR TITLE
1 openhim configuration

### LINKED ISSUE #1: Define OpenHIM standard configuration to be used for TWG 2.0
Define and share OpenHIM (see [OpenHIM Core Component](https://github.com/jembi/openhim-core-js)) standard configuration for TWG 2.0 project. Participating CHISs need to agree is correct, accurate and inclusive of required features, in the context of  Loss to Follow Up workflow.

Subtasks:

- [ ] Initial commit with Instant OpenHIE's core package
- [ ] Define FHIR resources required
- [ ] Add eslint to the mediator directory
- [ ] Run all services require with one command

### PR DESCRIPTION
**Pull request by samuelimoisili**
### Description
- Added a `startup.sh` script to help automate the startup and shutdown process.
- Added a `configurator` which sets up the `user`, `client` and `channel` details on OpenHIM. It also registers a `channel` for `hapi-fihr`.
- Removed the bloated OpenHIE instant docker setup and replaced it with a leaner docker compose setup.

Closes #1

### CODE REVIEW

File: configurator/libs/generators.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+const {generatePassword} = require('../utils');
+
+const CLIENT_ROLES = ['interop'];
+
+async function generateClient (password) {
+  const {passwordSalt, passwordHash, passwordAlgorithm} =
+    await generatePassword(password);
```
[the remark this answers is not in the document]
Comment (samuelimoisili) [reply configurator/libs/generators.js:7]: If for some reason `generatePassword` fails, I don't think we want it to fail gracefully.

File: package.json [hunk 2]
```diff
@@ -4,8 +4,11 @@
   "description": "This is a placeholder for 2023 interoperability project",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "lint": "eslint . --fix"
+    "lint": "eslint . --fix",
+    "init": "./startup.sh init",
```
[the remark this answers is not in the document]
Comment (samuelimoisili) [reply package.json:8]: They're development commands, I don't think it's important for getting started. We could add them when creating a more detailed guide.

**Comment by andrablaj** [comment]
@samuelimoisili @njogz can you please rename this repo to `interoperability`, as it is not a placeholder anymore? Thanks!

File: README.md [hunk 3]
```diff
@@ -1,2 +1,18 @@
-# interoperability-placeholder
+# Interoperability
 This is a placeholder for 2023 interoperability project 
+
+## Get Started
+
+1. Run `./startup.sh init` to start-up the docker containers on the first run or after calling `./startup.sh destroy`. Use `./startup.sh up` for subsequent runs after calling `init` without calling `destory`.
```
Comment (lorerod) [inline README.md:6]:
```suggestion
1. Run `./startup.sh init` to start-up the docker containers on the first run or after calling `./startup.sh destroy`. Use `./startup.sh up` for subsequent runs after calling `init` without calling `destroy`.
```

Comment (lorerod) [reply README.md:6]: Please summarize what will be created with this command.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,18 @@
...
+
+1. The default User username for OpenHIM is `interop@openhim.org` and password is `interop-password`. The default Client username is `interop-client` and password is `interop-password`.
+
+1. `./startup.sh down` or `./startup.sh destroy` to shutdown the servers.
+
+1. Visit https://localhost:9000 and login with the following credentials email: `interop@openhim.org` and password: `interop-password`.
+
+1. Once logged in, visit https://localhost:9000/#!/mediators and select the only mediator with the `Name` 'mediator'.
```
Comment (lorerod) [inline README.md:14]: I'm seeing a mediator with the `Name` 'Loss to Follow Up Mediator'

Comment (lorerod) [reply README.md:14]: Captura de pantalla 2023-01-13 a la(s) 14 22 43

File: README.md [hunk 5]
```diff
@@ -1,2 +1,18 @@
...
+
+1. Select the green `+` button to the right of the default channel to add the mediator.
+
+1. You can test the mediator by running `curl -X GET http://localhost:5001/scaffold -H "Authorization: Basic $(echo -n interop-client:interop-password | base64)"`. You should get the following response `{"status": "success" }`
```
Comment (lorerod) [inline README.md:18]: I'm getting `HTTP/1.1 404 Not Found` 
And can see this in openhim-core logs:
```
info: interop-client could NOT be authenticated, trying next auth mechanism if any...
info: Could NOT authenticate via TLS: undefined, trying next auth mechanism if any...
info: No channel matched the request /scaffold
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,18 @@
-# interoperability-placeholder
+# Interoperability
 This is a placeholder for 2023 interoperability project
```
Comment (lorerod) [inline README.md:2]: Please summarize what is this repository about

Comment (lorerod) [inline README.md:2]:
```suggestion

```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,18 @@
...
+
+1. `./startup.sh down` or `./startup.sh destroy` to shutdown the servers.
+
+1. Visit https://localhost:9000 and login with the following credentials email: `interop@openhim.org` and password: `interop-password`.
+
+1. Once logged in, visit https://localhost:9000/#!/mediators and select the only mediator with the `Name` 'Loss to Follow Up Mediator'.
+
+1. Select the green `+` button to the right of the default channel to add the mediator.
+
+1. You can test the mediator by running `curl -X GET http://localhost:5001/mediator -H "Authorization: Basic $(echo -n interop-client:interop-password | base64)"`. You should get the following response `{"status": "success" }`
```
Comment (lorerod) [inline README.md:18]: @samuelimoisili when running the curl command I'm getting:
`curl: (3) URL using bad/illegal format or missing URL`
And openhim-core logs shows:
```
2023-01-15 11:24:04 2023-01-15T14:24:04.404Z [worker1] info: starting job: generate transaction alerts, Last Ran at: Sun Jan 15 2023 14:24:04 GMT+0000 (Coordinated Universal Time)
2023-01-15 11:24:04 2023-01-15T14:24:04.424Z [worker1] info: Job generate transaction alerts has completed
2023-01-15 11:24:04 2023-01-15T14:24:04.429Z [worker1] info: starting job: auto retry failed transactions, Last Ran at: Sun Jan 15 2023 14:24:04 GMT+0000 (Coordinated Universal Time)
2023-01-15 11:24:04 2023-01-15T14:24:04.435Z [worker1] info: Job auto retry failed transactions has completed
2023-01-15 11:24:13 2023-01-15T14:24:13.233Z [worker1] info: No basic auth details supplied, trying next auth mechanism if any...
2023-01-15 11:24:13 2023-01-15T14:24:13.234Z [worker1] info: Could NOT authenticate via TLS: undefined, trying next auth mechanism if any...
2023-01-15 11:24:13 2023-01-15T14:24:13.241Z [worker1] info: The channel that matches the request /mediator is: Mediator
2023-01-15 11:24:13 2023-01-15T14:24:13.242Z [worker1] info: The request, '/mediator', is not authorised to access any channels.
```

**Review by lorerod — CHANGES_REQUESTED** [review]
Thanks @samuelimoisili for considering my suggestions. I just have one dude with the instructions. I'm using a Mac with macOS 13.1 and Docker Desktop Version 4.15.0; Engine: 20.10.21; Compose: 1.29.2 and I'm not getting the `{"status": "success" }` response when running curl.

File: README.md (hunk 7 above)
Comment (lorerod) [reply README.md:18]: I also tried using the following command (generated by Postman): `curl -X GET -i http://localhost:5001/mediator -H 'authorization: Basic aW50ZXJvcC1jbGllbnQ6aW50ZXJvcC1wYXNzd29yZA==' -H 'cache-control: no-cache' -H 'postman-token: 0817aeb7-ea2d-f5d7-612b-38c6325a0b74'` and got: `HTTP 401 Unauthorized`
And openhim-core logs shows:
```
2023-01-15 12:25:58 2023-01-15T15:25:58.600Z [worker1] info: interop-client could NOT be authenticated, trying next auth mechanism if any...
2023-01-15 12:25:58 2023-01-15T15:25:58.601Z [worker1] info: Could NOT authenticate via TLS: undefined, trying next auth mechanism if any...
2023-01-15 12:25:58 2023-01-15T15:25:58.611Z [worker1] info: The channel that matches the request /mediator is: Mediator
2023-01-15 12:25:58 2023-01-15T15:25:58.611Z [worker1] info: The request, '/mediator', is not authorised to access any channels.
2023-01-15 12:26:03 2023-01-15T15:26:03.620Z [worker1] info: HTTP socket timeout reached
```

**Review by lorerod — APPROVED** [review]
Thanks, @samuelimoisili. I finally got this working locally with the instructions.

**Pull request merged**